enhancements/machine-api/spot-instances.md

```diff
@@ -0,0 +1,359 @@
+---
+title: spot-instances
+authors:
+  - "@JoelSpeed"
+  - "@enxebre"
+reviewers:
+  - "@enxebre"
+  - "@bison"
+  - "@michaelgugino"
+approvers:
+  - "@enxebre"
+  - "@bison"
+  - "@michaelgugino"
+creation-date: 2020-02-04
+last-updated: 2020-02-04
+status: provisional
+see-also:
+replaces:
+superseded-by:
+---
+
+# Spot Instances
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+Enable OCP users to leverage cheaper, non-guaranteed instances to back Machine API Machines.
+
+## Motivation
+
+Allow users to cut costs of running OCP clusters on cloud providers by moving interruptible workloads onto non-guaranteed instances.
+
+### Goals
+
+- Provide consistent behaviour of Machines running on non-guaranteed instances across cloud providers
```

Comment:
This seems great as a goal...but I think it's going to be hard for anyone using this at scale to avoid needing to look at and understand provider-specific semantics and pricing.

Which, reading down seems to be the case (the provider pricing bits are all provider-specific). We're mostly just exposing the baseline infrastructure for integrating this nicely into OpenShift.

Maybe just call out as a goal something like

- Support sophisticated provider-specific automation driven by higher level (3rd party), e.g. code that dynamically adjusts price targets based on workload

Reply:
Yeah that's fair, I wrote the goals before doing most of the research 😅 Will reword